jberet-core/src/main/java/org/jberet/creation/ArchiveXmlLoader.java

```diff
@@ -109,55 +113,34 @@ public static Job loadJobXml(final String jobName, final ClassLoader classLoader
         return job;
     }
 
-    private static InputStream getJobXml(String jobXml, final ClassLoader classLoader) throws IOException {
+    private static InputStream getJobXml(String jobXml, final ClassLoader classLoader, final Iterable<JobXmlResolver> jobXmlResolvers) throws IOException {
         if (!jobXml.endsWith(".xml")) {
             jobXml += ".xml";
         }
-        // META-INF first
-        final String path = ARCHIVE_JOB_XML_DIR + jobXml;
-        InputStream is;
-        is = classLoader.getResourceAsStream(path);
-        if (is != null) {
-            return is;
-        }
 
-        // javax.jobpath system property. jobpath format?
-        File jobFile = null;
-        final String jobpath = WildFlySecurityManager.getPropertyPrivileged("javax.jobpath", null);
-        if (jobpath != null && !jobpath.isEmpty()) {
-            final String[] jobPathElements = jobpath.split(":");
-            for (final String p : jobPathElements) {
-                jobFile = new File(p, jobXml);
-                if (jobFile.exists() && jobFile.isFile()) {
-                    break;
-                }
-            }
-        }
-
-        // default location: current directory
-        if (jobFile == null) {
-            jobFile = new File(WildFlySecurityManager.getPropertyPrivileged("user.dir", "."), jobXml);
-            if (!jobFile.exists() || !jobFile.isFile()) {
-                //may be an absolute path to the job file
-                jobFile = new File(jobXml);
+        // Use the SPI to locate the job XML
+       for (JobXmlResolver jobXmlResolver : jobXmlResolvers) {
+            final InputStream is = jobXmlResolver.resolveJobXml(jobXml, classLoader);
+            if (is != null) {
+                return is;
             }
         }
-
-        is = new BufferedInputStream(new FileInputStream(jobFile));
-        return is;
+        throw BatchMessages.MESSAGES.failToGetJobXml(jobXml);
     }
 
     private static class JobXMLResolver implements XMLResolver {
```

Comment:
we should rename this inner class to avoid name confusion with the new one.  something like JobXMLSegmentResolver, or JobXMLExternalEntityResolver?